### DISCUSSION TITLE
why can't I merge my slices into named indeices in an object

### DISCUSSION THREAD

**Discussion by AhmedSarhanSwvl** [pmndrs/zustand#1025] [Q&A]
```jsx
const createBearSlice: StoreSlice<IBearSlice, IFishSlice> = (set, get) => ({
  eatFish: () =>
    set((prev) => ({ fishes: prev.fishes > 1 ? prev.fishes - 1 : 0 })),
});
const createRootSlice = (set: SetState<any>, get: GetState<any>) => ({
  bears: createBearSlice(set, get),
  fishes: createFishSlice(set, get),
});
```
when I fire `eatFish` function is doesn't fire and gives me the following error `TypeError: eatFishis not a function` 

when I go back to the example in the docs 
```jsx

const createRootSlice = (set: SetState<any>, get: GetState<any>) => ({
  ...createBearSlice(set, get),
  ...createFishSlice(set, get),
});
```
it does work fine,
is there a solution other than writing my action in the very top level of my store or making my global state data look like a train wreck

**Comment by dai-shi** [comment]
Thanks for using the library.

As you notice, only top level is merged.
For deeper levels, you need to merge by yourself, like this:
```js
set((prev) => ({ ...prev, fishes: prev.fishes > 1 ? prev.fishes - 1 : 0 })),
```

> making my global state data look like a train wreck

The rationale is that it's how flux is designed: https://github.com/pmndrs/zustand/wiki/Flux-inspired-practice

However, it's not only you who want namespacing. Here's the long issue: #178 
In short, it's open field for 3rd-party libraries and one promising lib is: [zustand-lens](https://github.com/dhmk083/dhmk-zustand-lens)
